app/services/classroom_config.rb

```diff
@@ -47,8 +48,65 @@ def configurable?(repo)
   # Returns nothing
   def generate_issues(repo, tree_sha)
     @github_repository.tree_objects(tree_sha).each do |issue|
-      blob = @github_repository.blob(issue.sha)
-      repo.create_issue(blob.data["title"], blob.body) if blob.data.present?
+      blob = formated_blob(@github_repository.blob(issue.sha))
+      next if blob.title.blank?
+      repo.create_issue(blob.title, blob.body, labels: blob.labels.present? ? blob.labels : [])
     end
   end
+
+  # Internal: Generates labels for the assignment_repository based on the configs
+  #
+  # repo - GitHubRepository for which to perform the configuration
+  #                   setups
+  # tree_sha     - sha of the "labels" tree
+  #
+  # Returns nothing
+  def generate_labels(repo, tree_sha)
+    delete_existing_labels(repo)
+    @github_repository.tree_objects(tree_sha).each do |label|
+      blob = formated_blob(@github_repository.blob(label.sha))
+
+      repo.create_label(blob.label, blob.color || "ffffff") if blob.label.present?
+    end
+  end
+
+  # Internal: Delete existing labels of a GitHubRepository
+  #
+  # repo - GitHubRepository
+  #
+  # Returns nothing
+  def delete_existing_labels(repo)
+    repo.labels.each do |label|
+      repo.delete_label! label[:name]
+    end
+  end
+
+  # Internal: Makes blob direct and data attributes accessible as method calls
+  #
+  # blob     - GitHubBlob to format
+  #
+  # Returns an OpenStruct of formated_blob instance
+  def formated_blob(blob)
+    fmt_blob = OpenStruct.new(body: blob.body, encoding: blob.encoding)
```

Comment:
Yeah, the benefit with this is that we have access to all blob data attributes as method calls. So we get to do e.i `blob.body`, `blob.labels` instead of `blob.body` `blob.data[:labels]`.